### PR TITLE
CSHARP-489 Move api docs to docfx

### PR DESCRIPTION
**Pull request by joao-r-reis**
DocFX has a nice template system where we can reference more than one template and it will merge them. I referenced one embedded template (`statictoc`) and one custom template. This custom template contains only the files that needed to be changed from the `statictoc` template so it's only a few files and I think it's fine if they are added to the repo. This allows anyone to build the docs (I added a `README` with instructions).

Preview:
imagem

### CODE REVIEW

**Comment by joao-r-reis** [comment]
Also I added a custom empty index page that only redirects to the API docs section but maybe we could add a simple index page with a small README or something... Maybe it would be a better user experience. (it's basically the same workaround that is being used in the C# Graph API Docs)

**Comment by jorgebay** [comment]
Would it be possible to move the template to documentor repo / doc team repo?

That way we can avoid having to maintain it along side our source code.

**Comment by joao-r-reis** [comment]
Done

File: doc/api-docs/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# How to build the C# Driver API Docs
```
Comment (jorgebay) [inline doc/api-docs/README.md]: Good idea to include the doc here.

**Pull request merged**